gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/commands/GoodbyeCommandHandler.java

```diff
@@ -49,8 +53,11 @@ public GoodbyeCommandHandler(InstallationService installationService) {
 
     @Override
     public Single<GoodbyeReply> handle(GoodbyeCommand command) {
-        return installationService.delete()
-                .andThen(Single.just(new GoodbyeReply(command.getId(), CommandStatus.SUCCEEDED)))
+        return installationService.getOrInitialize()
+                .map(Installation::getAdditionalInformation)
+                .doOnSuccess(additionalInfos -> additionalInfos.put(COCKPIT_INSTALLATION_STATUS, DELETED_STATUS))
+                .flatMap(installationService::setAdditionalInformation)
```

Comment:
Can we do all that in the delete() method of the installationService ? 

Not a huge fan of having a setter method at service level.

Reply:
I've added an `addAdditionalInformation` method on `InstallationService` and did the work here. I preferred not to change the `delete` method as the `GoodByeCommand` does not delete the installation anymore but instead just set the cockpit status to `DELETED`.